src/Sentry/Extensibility/DiagnosticLoggerExtensions.cs

```diff
@@ -68,17 +68,36 @@ internal static class DiagnosticLoggerExtensions
         public static void LogError<TArg>(
             this IDiagnosticLogger logger,
             string message,
-            Exception exception,
+            Exception? exception,
```

Comment:
Not supposed to be nullable.
We should have an overload that doesn't take it then